bus-mapping/src/circuit_input_builder/transaction.rs

```diff
@@ -195,7 +195,7 @@ pub struct Transaction {
     /// From / Caller Address
     pub from: Address,
     /// To / Callee Address
-    pub to: Address,
+    pub to: Option<Address>,
```

Comment:
The transaction struct had a bug where it would set the "to" field to a zero address when it was None. To fix this issue, I changed the tx's "to" field to Optional and made modifications to all the parts that use tx.to.
@chokobole @lightscale-luke @TomTaehoonKim